src/main/java/org/zeplinko/commons/lang/ext/core/Either.java

```diff
@@ -0,0 +1,70 @@
+package org.zeplinko.commons.lang.ext.core;
+
+import jakarta.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+@SuppressWarnings("LombokGetterMayBeUsed")
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class Either<L, R> {
+
+    private final L left;
+
+    private final R right;
+
+    public static <L, R> Either<L, R> left(@Nonnull L left) {
+        Objects.requireNonNull(left);
+        return new Either<>(left, null);
+    }
+
+    public static <L, R> Either<L, R> right(@Nonnull R right) {
```

Comment:
We need to explore ways how can we allow null values here.
Having a NonNull value here seems very limiting. Same like. Optional.of() which later required Optional.ofNullable()

But Allowing null here has challenges with regards to identification of Either type.
I have 3 approaches in mind

1. Having LeftEither and RightEither as subclasses and use instance type to identify the Either Type.
> Cons: This will have issues with Serialisation and Deserialisation into specific type. Especially for JSON deserialisation. 
2. Have a boolean flag to identify the either type
> Cons: This doesn't look very neat in terms of Serialised data. But the boolean flag will be a mandatory serialisation requirement.
3. Wrap the value inside the container object. Here we can use the Container is null vs Container has null differentiation. By far this seems to be most clean approach.

Thoughts?



4.